Source bash completion in picongpu profile

finnolec opened the pull request:
I don't have a lot of time left so I created this pullrequest preemptively. This should implement @ax3l suggestion in [this PR in the picongpu repo](https://github.com/ComputationalRadiationPhysics/picongpu/pull/3069), if it wasn't already fixed in [this pullrequest](https://github.com/ComputationalRadiationPhysics/picongpu/pull/3133/). I can't really test it myself because of time issues and I have no idea how spack works :D But I just copied and pasted the code from 3133 into the profile here as well, so it should theoretically work?

finnolec committed:
Source bash completion in picongpu profile

psychocoderHPC commented on packages/picongpu/picongpu.profile:
```diff
@@ -7,3 +7,11 @@ spack load @PIC_SPACK_COMPILER@ >/dev/null 2>&1 || echo "using default compiler"
+# activate bash completion if available
+BASH_COMP_FILE=$PICSRC/bin/picongpu-completion.bash
+if [ -f $BASH_COMP_FILE ] ; then
+    source $PICSRC/bin/picongpu-completion.bash
```
looks like I slept during adding this to PIConGPU. Could you please change this line to `source $BASH_COMP_FILE`

finnolec replied:
oops, and I didn't see it when I was copying it :D

finnolec committed:
Add @psychocoderHPC's suggestions

psychocoderHPC commented on packages/picongpu/picongpu.profile:
```diff
@@ -7,3 +7,11 @@ spack load @PIC_SPACK_COMPILER@ >/dev/null 2>&1 || echo "using default compiler"
+if [ -f $BASH_COMP_FILE ] ; then
+    source $BASH_COMP_FILE
+else
+    echo "bash completion file '$PICSRC/bin/picongpu-completion.bash' not found." >&2
```
If you have some more minutes this line can also be `echo "bash completion file '$BASH_COMP_FILE' not found." >&2`

finnolec replied:
I thought that this would mean, that the user sees the variable in the output and not the full path, so I just skipped it here. But I'm wrong and it should echo the full path. So I'll do it in a second :)

finnolec committed:
Implement @psychocoderHPC's suggestions

ax3l commented on packages/picongpu/picongpu.profile:
```diff
@@ -7,3 +7,11 @@ spack load @PIC_SPACK_COMPILER@ >/dev/null 2>&1 || echo "using default compiler"
 spack load -r @PIC_SPACK_SPEC@
+
+# activate bash completion if available
+BASH_COMP_FILE=$PICSRC/bin/picongpu-completion.bash
```
`BASH_COMP_FILE` is quite a generic name to set as environment var, did we merge this to PIConGPU? On the other hand its local and not exported...
Anyway, `PIC_COMP_FILE` would have been better to denote its something related to PIConGPU...

psychocoderHPC replied:
yes we merged it with this name.

ax3l replied:
@finnolec an update of the var name (which is exposed in the user environment) would be welcome :)

psychocoderHPC replied:
I changed it already in this PR.

ax3l commented on packages/picongpu/picongpu.profile:
```diff
@@ -7,3 +7,11 @@ spack load @PIC_SPACK_COMPILER@ >/dev/null 2>&1 || echo "using default compiler"
+if [ -f $BASH_COMP_FILE ] ; then
+    source $BASH_COMP_FILE
+else
+    echo "bash completion file '$BASH_COMP_FILE' not found." >&2
```
not sure if we should warn on this currently, only installs `@dev` will provide it so far and throw a confusing warning to all other (regular) users.

psychocoderHPC replied:
If we do not show this warning it means we should never advertise this feature.
If a user has issues and wonders why he can not use the auto-completion he maybe learned about in a workshop or the docu it is very hard to find the reason.

psychocoderHPC replied:
@ax3l could it be that you like to say we should skip this warning because in spack we currently only ship the master branch of picongpu but bash-completion files were only available in the dev branch?

ax3l replied:
What I mean is: in Spack we track some versions that are older and cannot have this file anyway and the `dev` branch that will always have this file.
Since the user can change little about this, I would not warn here. Basically newer releases have auto-completion, older releases don't.

psychocoderHPC committed:
rename bash variable and remove warning

- rename variable `BASH_COMP_FILE` to `PIC_COMP_FILE`
- remove warning if bash completion file is not available

psychocoderHPC commented:
@ax3l I pushed the changes to @finnolec branch. He leaved HZDR already :-( so I am not sure how often he is looking into github.

ax3l commented:
Thank you both! :)
